### PR TITLE
Aligned storage URL parsing with other platforms

### PR DESCRIPTION
**Pull request by amnguye**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Common/src/Shared/UriExtensions.cs [hunk 1]
```diff
@@ -89,15 +89,10 @@ public static string GetAccountNameFromDomain(string host, string serviceSubDoma
             var accountEndIndex = host.IndexOf(".", StringComparison.InvariantCulture);
             if (accountEndIndex >= 0)
             {
-                var serviceStartIndex = accountEndIndex + 1;
-                var serviceEndIndex = host.IndexOf(".", serviceStartIndex, StringComparison.InvariantCulture);
-                if (serviceEndIndex > serviceStartIndex)
+                var serviceStartIndex = host.IndexOf(serviceSubDomain, accountEndIndex, StringComparison.InvariantCulture);
+                if (serviceStartIndex > accountEndIndex)
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Common/src/Shared/UriExtensions.cs:93]: do we need this condition?

File: sdk/storage/Azure.Storage.Blobs/tests/BlobUriBuilderTests.cs [hunk 2]
```diff
@@ -220,24 +220,28 @@ public void BlobUriBuilder_RegularUrl_CNAME()
         }
 
         [Test]
-        public void BlobUriBuilder_RegularUrl_MalformedSubdomain()
+        public void DataLakeUriBuilder_AccountNamePeriod()
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Blobs/tests/BlobUriBuilderTests.cs:223]: Could you add test that we handle this case as well? https://github.com/Azure/azure-sdk-for-java/blob/5672b860b15aa32d1dd5b559f47c3cb1058f0166/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobUrlParts.java#L435-L436

We should also check that "azurite" urls parse correctly. Could you check if we have coverage like this in .NET https://github.com/Azure/azure-sdk-for-java/blob/5672b860b15aa32d1dd5b559f47c3cb1058f0166/sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/AzuriteTest.groovy#L48-L74 ? If not could you add it?

**Review by kasobol-msft — COMMENTED** [review]
LGTM.

File: sdk/storage/Azure.Storage.Common/src/Shared/UriExtensions.cs (hunk 1 above)
Comment (amnguye) [reply sdk/storage/Azure.Storage.Common/src/Shared/UriExtensions.cs:93]: You're right, I will take it out.

File: sdk/storage/Azure.Storage.Blobs/tests/BlobUriBuilderTests.cs (hunk 2 above)
Comment (amnguye) [reply sdk/storage/Azure.Storage.Blobs/tests/BlobUriBuilderTests.cs]: Should be done. There were also some existing tests that covered some of these cases.

File: sdk/storage/Azure.Storage.Common/src/Shared/UriExtensions.cs (hunk 1 above)
Comment (amnguye) [reply sdk/storage/Azure.Storage.Common/src/Shared/UriExtensions.cs]: done

**Review by kasobol-msft — APPROVED** [review]

**Review by seanmcc-msft — APPROVED** [review]

**Pull request merged**